src/devices/bus/rc2014/romram.cpp

```diff
@@ -25,19 +26,30 @@ class rom_ram_512k_device : public device_t, public device_rc2014_card_interface
 protected:
 	// device-level overrides
 	virtual void device_start() override;
-	virtual void device_post_load() override { update_banks(); }
+	virtual void device_reset() override;
+	virtual void device_add_mconfig(machine_config &config) override;
 	virtual const tiny_rom_entry *device_rom_region() const override;
 
-	void page_w(offs_t offset, uint8_t data) { m_page_reg[offset & 3] = data & 0x3f; update_banks(); }
-	void page_en_w(offs_t, uint8_t data) { m_page_en = data & 1; update_banks(); }
+	void page_w(offs_t offset, uint8_t data) { m_page_reg[offset & 3] = data & 0x3f; }
+	void page_en_w(offs_t, uint8_t data) { m_page_en = data & 1; }
 
-	void update_banks();
+	void mem0_w(offs_t offset, uint8_t data) { mem_w(offset, data, 0); }
+	void mem1_w(offs_t offset, uint8_t data) { mem_w(offset, data, 1); }
+	void mem2_w(offs_t offset, uint8_t data) { mem_w(offset, data, 2); }
+	void mem3_w(offs_t offset, uint8_t data) { mem_w(offset, data, 3); }
+
+	uint8_t mem0_r(offs_t offset) { return mem_r(offset, 0); }
+	uint8_t mem1_r(offs_t offset) { return mem_r(offset, 1); }
+	uint8_t mem2_r(offs_t offset) { return mem_r(offset, 2); }
+	uint8_t mem3_r(offs_t offset) { return mem_r(offset, 3); }
 private:
+	void mem_w(offs_t offset, uint8_t data, uint8_t bank);
+	uint8_t mem_r(offs_t offset, uint8_t bank);
```

Comment:
This is a lot of glue that could be avoided if you’d made the bank a template parameter.